src/legacy/ui/public/agg_types/controls/switch.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { EuiSwitch, EuiToolTip, EuiSpacer } from '@elastic/eui';
+import { AggParamEditorProps } from 'ui/vis/editors/default';
+
+interface SwitchParamEditorProps extends AggParamEditorProps<boolean> {
+  dataTestSubj?: string;
+  displayLabel?: string;
+  displayToolTip?: string;
+  disabled?: boolean;
+}
+
+function SwitchParamEditor({
+  agg,
+  aggParam,
+  value,
+  setValue,
+  dataTestSubj,
+  displayToolTip,
+  displayLabel,
+  disabled,
+}: SwitchParamEditorProps) {
+  const { shouldShow } = aggParam;
+
+  if (shouldShow && !shouldShow(agg)) {
```

Comment:
This check has already been included in `AggParamReactWrapper`. It's no need to duplicate.